packages/core/src/layout/SignInPage/gitlabProvider.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { Grid, Typography, Button } from '@material-ui/core';
+import { InfoCard } from '../InfoCard/InfoCard';
+import { ProviderComponent, ProviderLoader, SignInProvider } from './types';
+import { useApi, gitlabAuthApiRef, errorApiRef } from '@backstage/core-api';
+
+const Component: ProviderComponent = ({ onResult }) => {
+  const gitlabAuthApi = useApi(gitlabAuthApiRef);
+  const errorApi = useApi(errorApiRef);
+
+  const handleLogin = async () => {
+    try {
+      const identity = await gitlabAuthApi.getBackstageIdentity({
+        instantPopup: true,
+      });
+
+      const profile = await gitlabAuthApi.getProfile();
+      onResult({
+        userId: identity?.id || profile.email,
```

Comment:
GitLab identity is not returning an id in profile Info, so we could use profile email or set the identity id somewhere else.

Reply:
This need to be done in the backend, as this identity needs to be proven by the backstage id token as well. This is for example how it's done for google auth atm: https://github.com/spotify/backstage/blob/2ff4071209a5d0713ee8082578611ae74f6f98d8/plugins/auth-backend/src/providers/google/provider.ts#L145

Reply:
Ok, I will do it there then. Just like that: `const id = profile.email.split('@')[0]; ` should work the same